lib/java/desktop-environment/src/main/java/org/enso/desktopenvironment/WindowsTrashBin.java

```diff
@@ -15,7 +15,7 @@
 import org.slf4j.LoggerFactory;
 
 @CContext(WindowsTrashBin.ShellApi.class)
-final class WindowsTrashBin implements TrashBin {
+final class WindowsTrashBin extends TrashBinFallback implements TrashBin {
```

Comment:
Using inheritance for implementation purposes not for any meaningful type hierarchy. What would [Lišková](https://en.wikipedia.org/wiki/Liskov_substitution_principle) said about it?

I would personally make `hardDeletePath` a static utility method in some existing class. But if you [hide all these classes from the Javadoc](https://github.com/enso-org/enso/pull/10440/files/20b892fd67eb8f9b945bbcef6b8d3d1497e37391..c5aee4a773e82939b42d34735c14e090af188016#r1693857544), then I don't care.